test/WebSites/RazorWebSite/Services/TestHeadTagHelperComponent.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+
+namespace RazorWebSite
+{
+    public class TestHeadTagHelperComponent : TagHelperComponent
+    {
+        public TestHeadTagHelperComponent()
+        {
+        }
+
+        public override int Order => 1;
+
+        public override Task ProcessAsync(TagHelperContext context, TagHelperOutput output)
+        {
+            if (context.TagName == "head")
+            {
+                var customAttribute = new TagHelperAttribute("inject");
+                context.AllAttributes.TryGetAttribute("inject", out customAttribute);
```

Comment:
When you want to know what the original attributes looked like doing `context.AllAttributes` is correct. When you want to see what the final output of the `TagHelper` was you'd use `output.Attributes`. Both can be correct here so choose which one you believe in more 😄.

In the end lets do `[All]Attributes.ContainsName("inject")`